skbio/metadata/tests/test_interval.py

```diff
@@ -0,0 +1,568 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import unittest
+
+from skbio.metadata._interval import (_assert_valid_interval,
+                                      _assert_valid_boundary)
+from skbio.metadata import Interval
+from skbio.metadata import IntervalMetadata
+
+
+class TestInterval(unittest.TestCase):
+
+    def test_default_constructor(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(True, True), (True, True)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_constructor(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=[(1, 2), (4, 7)],
+                     boundaries=[(True, False), (False, False)],
+                     metadata={'name': 'sagA', 'function': 'transport'})
+
+        self.assertTrue(f._interval_metadata is not None)
+        self.assertListEqual(f.intervals, [(1, 2), (4, 7)])
+        self.assertListEqual(f.boundaries, [(True, False), (False, False)])
+        self.assertDictEqual(f.metadata, {'name': 'sagA',
+                                          'function': 'transport'})
+
+    def test_constructor_iterables(self):
+        f = Interval(interval_metadata=IntervalMetadata(),
+                     intervals=((1, 2), (4, 7)),
```

Comment:
given the name of the test, can `iter` be called? Right now this is the same as the test on line 32 except its using a `tuple` instead of a `list`, but I think it would be improve the test to operate over a generator